### PR TITLE
importccl: set up benchmark for import default expression-related operations

### PR DESCRIPTION
**Pull request by Anzoteh96**
This PR is for slide-making purpose. Please have a look to see if I've done things correctly. 

There's no plan to merge unless reviewers suggest me otherwise. 

Release note: None

### CODE REVIEW

**Review by Anzoteh96 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52886)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/ccl/importccl/import_stmt_test.go, line 3359 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52886#-MEu7ZuB3f7wxpJVD_YC:-MEu7ZuB3f7wxpJVD_YD:b-biorae) ([raw file](https://github.com/cockroachdb/cockroach/blob/802993ff2fc41ef0bd660dbe3bbcea2343c95667/pkg/ccl/importccl/import_stmt_test.go#L3359)):*
> ```Go
> // BenchmarkImportDefault/default-constant-function-8     	1000000000	         0.335 ns/op
> // BenchmarkImportDefault/default-now-8                   	1000000000	         0.528 ns/op
> // BenchmarkImportDefault/default-rowid-8                 	1000000000	         0.718 ns/op
> ```

Notes on these figures: the first two correspond to two columns a, b, with no non-targeted default columns. The remainder correspond to three columns: a, b are from the CSV file; c is non-targeted default columns. From these figures it seems like: 

1. For constant functions we basically get the default column for free (runtime-wise speaking); 
2. For now() it costs significantly more, partly because the need to (re)-evaluate the expressions each row instead of caching at the row converter; 
3. rowid() is even more expensive due to the overloading of function.

**Review by miretskiy — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52886)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @Anzoteh96)

___
*[pkg/ccl/importccl/import_stmt_test.go, line 3432 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52886#-MEwPPNXEaPRszXrlXOf:-MEwPPNY1jRy3JSkGhP5:btuqaqn) ([raw file](https://github.com/cockroachdb/cockroach/blob/802993ff2fc41ef0bd660dbe3bbcea2343c95667/pkg/ccl/importccl/import_stmt_test.go#L3432)):*
> ```Go
> 			defer sqlDB.Exec(b, `DROP TABLE t`)
> 			sqlDB.Exec(b, fmt.Sprintf(`CREATE TABLE t (%s)`, test.create))
> 			// Try importing 3 times within each test.
> ```

why?  you should be running benchmark b.N number of times.  Otherwise, your numbers are not really correctly

**Review by Anzoteh96 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52886)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy)

___
*[pkg/ccl/importccl/import_stmt_test.go, line 3432 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52886#-MEwPPNXEaPRszXrlXOf:-MExOXKaCIYgBewFAJ3B:bys7w4h) ([raw file](https://github.com/cockroachdb/cockroach/blob/802993ff2fc41ef0bd660dbe3bbcea2343c95667/pkg/ccl/importccl/import_stmt_test.go#L3432)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

why?  you should be running benchmark b.N number of times.  Otherwise, your numbers are not really correctly
</blockquote></details>

Just did the change now. Is that the correct way of doing this?

File: pkg/ccl/importccl/import_stmt_test.go [hunk 1]
```diff
@@ -3377,6 +3377,130 @@ func TestImportDefault(t *testing.T) {
...
+			targetCols: "a, b",
+			format:     "CSV",
+		},
+	}
+	for _, test := range testCases {
+		b.Run(test.name, func(b *testing.B) {
+			defer sqlDB.Exec(b, `DROP TABLE t`)
+			sqlDB.Exec(b, fmt.Sprintf(`CREATE TABLE t (%s)`, test.create))
+			b.ResetTimer()
+			// Try importing 3 times within each test.
```
Comment (pbardea) [inline pkg/ccl/importccl/import_stmt_test.go:3489]: I think this comment is now outdated?

**Review by pbardea — APPROVED** [review]
These LGTM

File: pkg/ccl/importccl/import_stmt_test.go (hunk 1 above)
Comment (Anzoteh96) [reply pkg/ccl/importccl/import_stmt_test.go]: Yes it is, just removed :)

**Comment by Anzoteh96** [comment]
Closing this given that there's no intention to merge. Thanks for reviewing and correcting the benchmarking mistakes I may have made!

**Comment by miretskiy** [comment]
btw, your numbers include the time to drop table (due to defer statement)
-- not sure how bad it is in terms of making your #s worse than they might
be.

**Comment by Anzoteh96** [comment]
> btw, your numbers include the time to drop table (due to defer statement) -- not sure how bad it is in terms of making your #s worse than they might be.
> […](#)

Ah, good point. I should have considered that. :)

**Pull request closed without merging**